### PR TITLE
Fixed responses for request errors in benchmarks

### LINKED ISSUE #20: Request errors breaks response tab
When the request handler class detects the response status code not matching the expected one it returns the error code and message, but doesn't include the response so the tab displays undefined.

https://github.com/matteofigus/api-benchmark/blob/master/lib/request-handler.js

I'm wondering what the best way to fix this is. I also noticed if you pass in a string it doesn't match since it is doing strict comparison to of them and expects ints. We could do parseInt() if we want to support passing in a string.

screen shot 2015-06-02 at 4 52 44 pm
screen shot 2015-06-02 at 4 52 51 pm

### PR DESCRIPTION
**Pull request by dmyers**
Played around with this today. This fixes #20 with any error with requests like status code or mean times.

Does this look good @matteofigus?

### CODE REVIEW

File: lib/request-handler.js [hunk 1]
```diff
@@ -8,27 +8,29 @@ module.exports = {
   make: function(req, suite, suiteName, requestAgent, callback){
     requestAgent.make(req, function(err, response){
 
+      var res = {
```
Comment (matteofigus) [inline lib/request-handler.js]: I would check if response is null here.
```js
var res = !!response ? { ...} : false;
```

**Comment by dmyers** [comment]
That's a good idea! Updated the PR.

**Comment by dmyers** [comment]
That's a good idea! Updated the PR. I'm not sure why Appveyor build is failing, but updated tests for Travis.

**Comment by dmyers** [comment]
If you think this is a good solution and we end up merging, it would be great to get a NPM update too. I've got an email report I generate from benchmarking some APIs and send with a list of routes with errors and slow routes and so with this new addition I will add the response status code to them to say why they are considered "broken".

**Comment by matteofigus** [comment]
Ok. Problem with AppVeyor is that there are a couple of tests that should be re-written as they are instable. This is separate, so let's merge this.

**Comment by matteofigus** [comment]
Released on npm: `0.2.3`

**Comment by dmyers** [comment]
Thanks!

**Pull request merged**